django/test/testcases.py

```diff
@@ -1273,6 +1275,10 @@ def terminate(self):
             self.httpd.server_close()
 
 
+def is_connection_sqlite_in_memory(conn):
```

Comment:
Yes and no. It would make sense to move the part after the "and" to `sqlite`'s version of `DatabaseWrapper` since only that version has the `is_in_memory_db()` method. But there is still the logic of checking that the backend is `sqlite`, which it doesn't seem like we'd want to put in the base implementation.

So I think we'd still want a helper function somewhere to do both the sqlite check and the in-memory check (though I'm not sure where is best -- in one of the TestCase classes?). I started thinking about these issues before, but I wasn't sure if it was straying too far from the purpose of the patch. I'm happy to include more refactoring if it's acceptable.